### DISCUSSION TITLE
When I click on "Send to Obsidian" for an already existing note, it overwrites all tags and the content.

### DISCUSSION THREAD

**Discussion by Developer-42** [coddingtonbear/obsidian-web#218] [General]
I don't see this as a bug, but I imagine it gets tedious quickly if you want to update "snapshot" notes to the current webpage as all your manually inserted tags will be lost, just like anything you added at the end of the content, for example.

So in my case, `sourceAuthor` and `publishDate` are two properties in Obsidian that I have to enter manually, and it would overwrite them back to empty.

Is there a fix for this?

Here is the content template (_where `publishDate` is not tied to a specific day_):
```---
tags:
  - Snapshot
title: {{json page.title}}
sourceType: Webpage
sourceURL: {{page.url}}
sourceAuthor:
  - {{author}}
publishDate: 
snapshotDate: "{{date}}"
---
{{#if page.selectedText}}

{{quote page.selectedText}}

---

{{/if}}{{page.content}}

**Comment by coddingtonbear** [comment]
I'm afraid there isn't a great solution for what you're encountering right now, but you might have some more options after https://github.com/coddingtonbear/obsidian-local-rest-api/issues/26 lands.

The core problem is that the default "Capture page snapshot" option (and all functions of this extension) interact with Obsidian via [Obsidian Local REST API](https://github.com/coddingtonbear/obsidian-local-rest-api), and that particular default option does so via the `PUT` API defined here: https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Files/put_vault__filename_ .  You can see that first hand if you open the extension options and try to edit that template.  Unfortunately, that `PUT` API isn't intended for any sort of partial update like what you seem to be hoping for.

Back to the hopeful news, though: if you look around on that same documentation I linked above you'll see that there *is* an API  that's intended to be used for doing partial updates -- https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Files/patch_vault__filename_.  Unfortunately, right now that API doesn't quite support what you're looking for -- it only supports inserting content relative to particular headers, but after https://github.com/coddingtonbear/obsidian-local-rest-api/issues/26 lands, the `PATCH` operation will be updated to provide a few more options, maybe including some kind of solution for the limitation you're encountering.
